### PR TITLE
expose path to gcloud folder in setup action

### PR DESCRIPTION
**Pull request by volatilemolotov**
expose CLOUDSDK_CONFIG path to kubelet storage on the node as runners use emptyDir. 
This is needed for jobs that run additional docker containers and are trying to mount gcloud folder from the node. This makes that possible when running ARC in GKE

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by volatilemolotov** [comment]
Run RAT PreCommit

**Comment by volatilemolotov** [comment]
Really not sure why RAT PreCommit is failing on post

**Comment by volatilemolotov** [comment]
assign set of reviewers

File: .github/actions/setup-action/action.yml [hunk 1]
```diff
@@ -57,3 +57,8 @@ runs:
         github_repository: ${{ github.repository }}     
         github_current_run_id: ${{ github.run_id }}
         pull_request_url: ${{ github.event.issue.pull_request.url }}
+    ### Used for Flink jobs that call docker run and mount gcloud folder
```
Comment (Abacn) [inline .github/actions/setup-action/action.yml:60]: The failure is due to setup-action.yml gets overwritten (by checkout action) during the workflow run, and parser running into inconsistency state. Likely a github issue.

----

If this is for certain flink job only, then we do not need it add to setup-action which every job run?

Comment (volatilemolotov) [reply .github/actions/setup-action/action.yml:60]: We decided to put it there as to hide the implementation from the people creating new workflows. 
Its not necessarily just for Flink but other jobs could use gcloud. 

Its not compute intensive so really not an issue to keep it there

Comment (volatilemolotov) [reply .github/actions/setup-action/action.yml:60]: > The failure is due to setup-action.yml gets overwritten (by checkout action) during the workflow run, and parser running into inconsistency state. Likely a github issue.

Should we ignore it or? Its unclear to me why it happens so im not sure how to apporach the fix

Comment (volatilemolotov) [reply .github/actions/setup-action/action.yml:60]: Removed the flink reference.

Ill try to look from time to time how the action behaves and is this something that is happening often.

**Comment by Abacn** [comment]
oops it looks like this breaks github action: https://github.com/apache/beam/actions/runs/5812126326/job/15756842872?pr=27934

**Pull request merged**